File: auth/src/main/java/com/harrish/auth/util/MessageResolver.java
package com.harrish.auth.util;

import org.springframework.context.MessageSource;
import org.springframework.context.i18n.LocaleContextHolder;
import org.springframework.stereotype.Component;

import java.util.Locale;

@Component
public class MessageResolver {

    private final MessageSource messageSource;

    public MessageResolver(MessageSource messageSource) {
        this.messageSource = messageSource;
    }

    public String getMessage(String code) {
        return getMessage(code, new Object[0]);
    }

    public String getMessage(String code, Object... args) {
        var locale = LocaleContextHolder.getLocale();

        return messageSource.getMessage(code, args, code, locale);
    }
}

File: auth-kotlin/src/main/kotlin/com/harrish/auth/util/MessageResolver.kt
package com.harrish.auth.util

import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder
import org.springframework.stereotype.Component
import java.util.Locale

@Component
class MessageResolver(
    private val messageSource: MessageSource
) {

    fun getMessage(code: String): String {
        return getMessage(code, emptyArray())
    }

    fun getMessage(code: String, vararg args: Any): String {
        val locale: Locale = LocaleContextHolder.getLocale()
        return messageSource.getMessage(code, args, code, locale)
    }
}
